tools/snp.sh

```diff
@@ -268,10 +319,110 @@ install_dependencies() {
+      echo "set environment variable RHEL_SUBS_MGR_PASS to RedHat Portal Password"
+      flag=1
+  fi
+  return $flag
+}
+
+rhel_subscription_mgr_register(){
+  local subscription_manager_status=$(sudo subscription-manager status| grep "Overall Status" )
+
+  # Add different keyterms for RedHat system unregister status like unknown, unregister, etc.
+  local different_words_for_unregister=( "unknown" "unregister"  )
+
+  # Convert to lowercase for case in-sesitive match
+  local subscription_manager_status=${subscription_manager_status,,}
+
+  # Register only when the system subscription status is unregistered.
+  for each_word in "${different_words_for_unregister[@]}";do
+    case "${subscription_manager_status}" in
+      *"$each_word"*)
+
+        check_if_redhat_credentials_set
+
+        # Activate RedHat Subscription
+        sudo subscription-manager register --username ${RHEL_SUBS_MGR_USER} --password ${RHEL_SUBS_MGR_PASS} --force
+        break
+        ;;
+    esac
+  done
+}
+
+rhel_install_dependencies() {
+  # Step 1: Activate RedHat Subscription
+  rhel_subscription_mgr_register
+
+  # Build dependencies
+   # Group install on RH provisioned RH Host
+  sudo dnf groupinstall -y "Headless Management" "Development Tools" "Container Management" "RPM Development Tools"
+  sudo dnf install -y wget
+  sudo dnf install -y git
+  sudo dnf install -y make automake gcc gcc-c++ kernel-devel
+
+  # make dependencies
+  sudo dnf install -y libuuid-devel dwarves perl
+
+  # Enable RedHat Repository for qemu dependencies
+  sudo subscription-manager repos --enable codeready-builder-for-rhel-9-x86_64-rpms
+
+  # qemu dependencies
+  sudo dnf install -y ninja-build
+  sudo dnf install -y pkg-config
+  sudo dnf install  -y glib2-devel
+  sudo dnf install  -y pixman-devel
+  sudo dnf install -y libslirp-devel
+
+  # ovmf dependencies
+  sudo dnf install -y uuid-devel
+  sudo dnf install -y iasl
+
+  # Remove any default nasm package
+  sudo dnf remove nasm
+  install_nasm_from_source
+
+  # kernel dependencies
+  sudo dnf install -y rsync
+  sudo dnf install -y ncurses-devel
   
-  echo "true" > "${dependencies_installed_file}"
+  # libssl-dev is openssl-devel in RHEL
+  # rpm-build -- Scripts and executable programs used to build packages
+  sudo dnf install -y rpm-build
+
+  # cloud-utils dependency
+  sudo dnf install -y cloud-init
+
+  # sev-snp-measure
+  sudo dnf install -y python3-pip
+
+  sudo dnf install -y curl
+
+  # Handles grub menu config
+  sudo dnf info -y grubby
+
+  # genisoimage for cloud-init data(VM user-data and metadata) seed
+    # genisoimage utility supported until RedHat version 8
+    # For genisoimage utility support in RedHat 9, using epel
+  sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
+  sudo dnf install -y genisoimage
+
+  sudo dnf install -y curl
+
+  # Installing additional dependencies on RHEL Host provisioned from satellite
+  sudo dnf install -y bc wget
 }
 
+
```

Comment:
Done.. I cleaned this up

Reply:
Since we have an install.sh in AMDSEV, can we move the installation of tools like Rust, NASM, and other dependencies from snp.sh into this file? As we continue to support more distributions, snp.sh is becoming more complicated. Please check this: https://github.com/ramagali24/AMDSEV-SVSM/blob/svsm-latest-fixes/install.sh, which includes Fedora and openSUSE. @ryansavino, I wanted to discuss this with you. What do you suggest?